src/Language/PureScript/AST/Exported.hs

```diff
@@ -144,3 +141,16 @@ isDctorExported ident (Just exps) ctor = test `any` exps
   test (TypeRef _ ident' Nothing) = ident == ident'
   test (TypeRef _ ident' (Just ctors)) = ident == ident' && ctor `elem` ctors
   test _ = False
+
```

Comment:
Results for compiling https://github.com/hdgarrood/solitaire from scratch on current master vs this branch, n=10:

```
master:
mean: 15.907499999999999s
std dev: 0.3209788933870889s
reordered:
mean: 15.8984s
std dev: 0.29505531684753594s
```

I chose that repo because it depends on halogen, css, and dom, all of which contain some fairly large modules with lots of exports and re-exports.

Note that the std dev is much larger than the difference between the means, so there's no evidence to suggest this change has had any performance effect.

Full results at https://gist.github.com/hdgarrood/963eb75bc566074300e438996eeda11e